src/timer.rs

```diff
@@ -0,0 +1,384 @@
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tier $channel>].modify(|_r, w| w.tier_2().set_bit());
+                }
+
+                /// Disable interrupt for match register 0.
+                pub fn disable_match0_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tier $channel>].modify(|_r, w| w.tier_0().clear_bit());
+                }
+
+                /// Disable interrupt for match register 1.
+                pub fn disable_match1_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tier $channel>].modify(|_r, w| w.tier_1().clear_bit());
+                }
+
+                /// Disable interrupt for match register 2.
+                pub fn disable_match2_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tier $channel>].modify(|_r, w| w.tier_2().clear_bit());
+                }
+
+                /// Enable this counter
+                pub fn enable(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.tcer.modify(|_r, w| w.[<timer $channel _en>]().set_bit());
+                }
+
+                /// Disable this counter
+                pub fn disable(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.tcer.modify(|_r, w| w.[<timer $channel _en>]().set_bit());
+                }
+
+                /// Clear interrupt for match register 0.
+                pub fn clear_match0_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_0().set_bit());
+                }
+
+                /// Clear interrupt for match register 1.
+                pub fn clear_match1_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_1().set_bit());
+                }
+
+                /// Clear interrupt for match register 2.
+                pub fn clear_match2_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_2().set_bit());
+                }
+
+                /// Sets when the to preload.
+                pub fn set_preload(&self, preload: Preload) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer
+                        .[<tplcr $channel>]
+                        .modify(|_r, w| unsafe { w.tplcr().bits(preload.to_prlcr()) });
+                }
+
+                /// Sets match register 0
+                pub fn set_match0(&self, time: Milliseconds) {
+                    let time = (self.clock.0 as u128 * time.0 as u128 / 1_000u128) as u32;
```

Comment:
128bit math for the match values seems a bit excessive.
64bit would still allow the timer to get to the max value of u32